File: android/app/src/main/java/io/github/martinschneider/juvavum/activity/RulesActivity.java
package io.github.martinschneider.juvavum.activity;

import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.text.Html;
import android.text.method.ScrollingMovementMethod;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.TextView;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import io.github.martinschneider.juvavum.R;

public class RulesActivity extends AppCompatActivity {

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_settings:
                startActivity(new Intent(RulesActivity.this, SettingsActivity.class));
                return true;
            case R.id.action_game_controls:
                startActivity(new Intent(RulesActivity.this, GameControlsActivity.class));
                return true;
            case R.id.action_about:
                startActivity(new Intent(RulesActivity.this, AboutActivity.class));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rules);
        TextView textView = findViewById(R.id.textView);
        textView.setText(Html.fromHtml(getString(R.string.help_text), new ImageGetter(), null));
        textView.setMovementMethod(new ScrollingMovementMethod());
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setTitle(R.string.title_activity_rules);
        }
    }

    private class ImageGetter implements Html.ImageGetter {

        public Drawable getDrawable(String source) {
            int id = RulesActivity.this.getResources().getIdentifier(source, "drawable", RulesActivity.this.getPackageName());
            Drawable d = getResources().getDrawable(id);
            d.setBounds(0, 0, d.getIntrinsicWidth(), d.getIntrinsicHeight());
            return d;
        }
    }

}

File: android/app/src/main/java/io/github/martinschneider/juvavum/activity/RulesActivity.kt
package io.github.martinschneider.juvavum.activity

import android.content.Intent
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.text.Html
import android.text.method.ScrollingMovementMethod
import android.view.Menu
import android.view.MenuItem
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import io.github.martinschneider.juvavum.R

class RulesActivity : AppCompatActivity() {
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_settings -> {
                startActivity(Intent(this@RulesActivity, SettingsActivity::class.java))
                true
            }
            R.id.action_game_controls -> {
                startActivity(Intent(this@RulesActivity, GameControlsActivity::class.java))
                true
            }
            R.id.action_about -> {
                startActivity(Intent(this@RulesActivity, AboutActivity::class.java))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rules)
        val textView = findViewById<TextView>(R.id.textView)
        textView.text = Html.fromHtml(getString(R.string.help_text), ImageGetter(), null)
        textView.movementMethod = ScrollingMovementMethod()
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setTitle(R.string.title_activity_rules)
        }
    }

    private inner class ImageGetter : Html.ImageGetter {
        override fun getDrawable(source: String): Drawable {
            val id = this@RulesActivity.resources.getIdentifier(source, "drawable", this@RulesActivity.packageName)
            val d = resources.getDrawable(id)
            d.setBounds(0, 0, d.intrinsicWidth, d.intrinsicHeight)
            return d
        }
    }
}